environment/src/test/java/com/sequenceiq/environment/environment/flow/creation/handler/ResourceEncryptionInitializationHandlerTest.java

```diff
@@ -137,6 +158,42 @@ void acceptTestEnvironmentShouldNotBeUpdatedWhenEncryptionKeyUrlIsEmpty() {
         verifyEnvCreationEvent();
     }
 
+    @Test
+    void acceptTestEnvironmentShouldBeUpdatedWhenEncryptionKeyUrlIsPresent() {
+        doAnswer(i -> null).when(eventSender).sendEvent(baseNamedFlowEventCaptor.capture(), any(Headers.class));
+        Environment environment = new Environment();
+        environment.setParameters(newAzureParameters());
+        when(environmentService.findEnvironmentById(ENVIRONMENT_ID)).thenReturn(Optional.of(environment));
+        CreatedDiskEncryptionSet createdDiskEncryptionSet = new CreatedDiskEncryptionSet.Builder()
+                .withDiskEncryptionSetId("dummmy-set-id")
+                .build();
+        when(environmentEncryptionService.createEncryptionResources(any(), any())).thenReturn(createdDiskEncryptionSet);
+
+        underTest.accept(environmentDtoEvent);
+
+        verify(eventSender).sendEvent(baseNamedFlowEventCaptor.capture(), headersArgumentCaptor.capture());
+        AzureParameters azureParameters = (AzureParameters) environment.getParameters();
+        assertEquals(azureParameters.getDiskEncryptionSetId(), "dummmy-set-id");
+        assertEquals(environment.getStatus(), EnvironmentStatus.ENVIRONMENT_ENCRYPTION_RESOURCES_INITIALIZED);
```

Comment:
What happened to the `environmentService.save()` invocation? Could you please add a verification (with an `ArgumentCaptor`) like I asked earlier?

> Instead of simply verifying the act of the `save()` invocation, please capture the method argument and then check if the DESid inside the captured Environment matches the one returned above.

Reply:
To line 176: It is the 3rd time I am repeating this: do not just verify the act of `save()` invocation, but also capture its actual argument & check its expected contents. Moreover, the `, times(1)` part shall be dropped, that is the default invocation count if not specified.

Reply:
Notes to line 176:

```
verify(environmentService, times(1)).save(environment);
```

1. Mockito matches method arguments using object equality based on `equals()` by default (i.e. as if `.save(eq(environment))`); see [the documentation](https://javadoc.io/static/org.mockito/mockito-core/3.9.0/org/mockito/Mockito.html#argument_matchers). This check only works because the `Environment` and `AzureParameters` entity classes do not currently override `equals()`, hence falling back to the default implementation based on reference equality. In other words: Since, the contents of `environment` is expected to change after executing `underTest.accept(environmentDtoEvent);` (notably `environment.parameters.diskEncryptionSetId` will be updated), this check will break as soon as someone adds a custom `equals()` implementation to `Environment` and `AzureParameters`. My original suggestion was to use an explicit `ArgumentCaptor` to capture the argument of the `save()` invocation, that would be immune to any changes regarding the `equals()` implementations of the above classes.
2. `, times(1)` is redundant because 1 is the default repetition count.